indico/modules/events/persons/client/js/Email.jsx

```diff
@@ -0,0 +1,264 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2022 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import emailAttributesURL from 'indico-url:persons.email_event_persons';
+import emailPreviewURL from 'indico-url:persons.email_event_persons_preview';
+
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import {FormSpy} from 'react-final-form';
+import {Form, Button, TextArea, Message, Dimmer, Loader} from 'semantic-ui-react';
+
+import PlaceholderInfo from 'indico/react/components/PlaceholderInfo';
+import TextEditor, {FinalTextEditor} from 'indico/react/components/TextEditor';
+import {FinalCheckbox, FinalDropdown, FinalInput, handleSubmitError} from 'indico/react/forms';
+import {FinalModalForm} from 'indico/react/forms/final-form';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {Param, Plural, PluralTranslate, Singular, Translate} from 'indico/react/i18n';
+import {indicoAxios} from 'indico/utils/axios';
+import {snakifyKeys} from 'indico/utils/case';
+
+import './Email.module.scss';
+
+const useIdSelector = selector =>
```

Comment:
That's not a hook, please don't use `useWhatever` naming for "normal" utils.

Also, why is sizing information relevant to getting selected IDs? Is this to skip hidden checkboxes?

Reply:
Yep. As far as I can tell this is only used when a filter is applied to the author list.
Maybe I should add a comment there, since this check might become pointless once author lists are converted to React (which I think it was talked about)?